client/state/data-layer/wpcom/sites/settings/index.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Internal dependencies
+ */
+import {
+	SITE_SETTINGS_SAVE,
+	SITE_SETTINGS_SAVE_FAILURE,
+	SITE_SETTINGS_SAVE_SUCCESS,
+	SITE_SETTINGS_REQUEST,
+	SITE_SETTINGS_REQUEST_FAILURE,
+	SITE_SETTINGS_REQUEST_SUCCESS,
+} from 'state/action-types';
+import { receiveSiteSettings, updateSiteSettings } from 'state/site-settings/actions';
+import { normalizeSettings } from 'state/site-settings/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+
+export const fetchSiteSettings = ( { dispatch }, action ) => {
+	const { siteId } = action;
+
+	dispatch(
+		http(
+			{
+				method: 'GET',
+				path: `/sites/${ siteId }/settings`,
+				apiVersion: '1.1',
+			},
+			action
+		)
+	);
+};
+
+export const addSiteSettings = ( { dispatch }, action, next, { name, description, settings } ) => {
+	const { siteId } = action;
+	const savedSettings = {
+		...normalizeSettings( settings ),
+		blogname: name,
+		blogdescription: description,
+	};
+	dispatch( receiveSiteSettings( siteId, savedSettings ) );
+	dispatch( {
+		type: SITE_SETTINGS_REQUEST_SUCCESS,
+		siteId,
+	} );
+};
+
+export const announceFetchFailure = ( { dispatch }, { siteId, meta } ) => {
+	dispatch( {
+		type: SITE_SETTINGS_REQUEST_FAILURE,
+		siteId,
+		error: meta.dataLayer.error,
+	} );
```

Comment:
it looks like the only purpose here is to indicate that the request is done. are we visually representing this to the customer? (I think we might be)

it would be awesome if we could get rid of this entirely and replace it with something that _announces_ or whatever. was this a copy/paste job? this action isn't really announcing…